pkg/providers/vsphere/vsphere.go

```diff
@@ -380,8 +409,10 @@ func (p *vsphereProvider) SetupAndValidateUpgradeCluster(ctx context.Context, cl
 		return fmt.Errorf("validating vsphere machine configs datastore usage: %v", err)
 	}
 
-	if err := p.validator.validateVsphereUserPrivs(ctx, vSphereClusterSpec); err != nil {
-		return fmt.Errorf("validating vsphere user privileges: %v", err)
+	if !p.skippedValidations[validations.VSphereUserPriv] {
+		if err := p.validator.validateVsphereUserPrivs(ctx, vSphereClusterSpec); err != nil {
+			return fmt.Errorf("validating vsphere user privileges: %v", err)
```

Comment:
Do we have an argument for not running a test case that triggers this error? It feels like the test case of verifying the error can be thrown would be an important one. Open to hearing counter arguments, those are just my first thoughts.

Reply:
No you are right. Just added the tests to cover it 👍